pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -318,8 +318,8 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 			case model.ProtocolTCP, model.ProtocolHTTPS, model.ProtocolMongo, model.ProtocolRedis:
 				listenerType = plugin.ListenerTypeTCP
 				if service.Resolution != model.Passthrough {
-					listenAddress = service.Address
-					addresses = []string{service.Address}
+					listenAddress = service.GetServiceAddressForNode(&node)
+					addresses = []string{listenAddress}
```

Comment:
why can't you provide all service addresses here?

Reply:
Any reason to do that?